src/praisonai-agents/praisonaiagents/tools/schema.py

```
@@ -149,4 +150,66 @@ def get_parameter_requirements(sig: inspect.Signature, param_name: str) -> bool:
+    schema = {
+        "type": "object",
+        "properties": {},
+        "required": []
+    }
+    
+    # Default skip set
+    if skip is None:
+        skip = {"self", "cls"}
+    
+    try:
+        for param_name, param in sig.parameters.items():
```

Ruff (0.15.17)

[warning] Loop control variable `param` not used within loop body

Rename unused `param` to `_param`

(B007)